### PR TITLE
Modify tag-retention UI

### PR DESCRIPTION
**Pull request by AllForNothing**
1. add disable/enable to retention rule.
2. add server paging
3. modify UI of add-rule component
#8521 
Signed-off-by: AllForNothing <sshijun@vmware.com>

### CODE REVIEW

File: src/portal/src/app/project/tag-retention/tag-retention.component.ts [hunk 1]
```diff
@@ -167,7 +169,34 @@ export class TagRetentionComponent implements OnInit {
         this.addRuleComponent.isAdd = false;
         this.ruleIndex = -1;
     }
-
+    disable(index) {
```
Comment (zhoumeina) [inline src/portal/src/app/project/tag-retention/tag-retention.component.ts]: this disable method and enable method can be refactor to one method with different parameters.

Comment (AllForNothing) [reply src/portal/src/app/project/tag-retention/tag-retention.component.ts]: fixed

File: src/portal/src/app/project/tag-retention/tag-retention.component.ts [hunk 2]
```diff
@@ -167,7 +169,20 @@ export class TagRetentionComponent implements OnInit {
         this.addRuleComponent.isAdd = false;
         this.ruleIndex = -1;
     }
-
+    disableOrEnable(index, isActionDisable) {
```
Comment (zhoumeina) [inline src/portal/src/app/project/tag-retention/tag-retention.component.ts]: better be named as toggleDisable()

Comment (AllForNothing) [reply src/portal/src/app/project/tag-retention/tag-retention.component.ts]: modified

**Pull request merged**